src/UI/GameBuilder.java

```diff
@@ -23,4 +30,21 @@ public void addCO(Commander co)
   {
     commanders.add(co);
   }
+
+  public GameInstance createGame()
+  {
+    GameInstance newGame = null;
+
+    // Build the CO list and the new map and create the game instance.
+    Commander[] cos = commanders.toArray(new Commander[commanders.size()]);
+    MapMaster map = new MapMaster( cos, mapInfo );
+    if( map.initOK() )
+    {
+      newGame = new GameInstance(map, isFowEnabled, defaultWeather, startingFunds);
+
+      for(Commander co: commanders) co.incomePerCity = incomePerCity;
```

Comment:
On second inspection, I'm realizing this violates how I had it in my head that COs should be set up.
My thought was that starting funds, income, ability energy exchange rates, and similar should be set at the time the CO is constructed, and not changed after.
Then you can put D2D effects in the constructor of a given CO and not worry about anybody else overwriting you.

Reply:
My initial idea was that those things would be stored statically, set during game init, and not changed... but that is a rather non-obvious way to design constructor arguments.
My counterproposal: GameBuilder gets the list of COInfos, and becomes an argument to the CO's constructor so game settings can be pulled in during init of the base CO.

Reply:
I'm not sure I like the idea of passing `GameBuilder` to `Commander` since that creates a new dependency for no tangible benefit.
If we really want these CO attributes to be set at construction and not changed after, we would have to make them `private` (or at least `protected`) and probably `final` as well.
Frankly, I'd be more inclined to make `incomePerCity` an attribute of `GameInstance` instead (or of a `Scenario` class that I think we will want eventually), since income is set per-game (not per-CO) anyway.

Reply:
> I'm not sure I like the idea of passing `GameBuilder` to `Commander` since that creates a new dependency for no tangible benefit.

In fact, these changes already break Bear&Bull.

> If we really want these CO attributes to be set at construction and not changed after, we would have to make them `private` (or at least `protected`) and probably `final` as well.

`protected` is fine with me (Though it will break our UI. We'll need a getter). `final` is not.
Bear&Bull set their income and unit costs to 90% in the constructor, but GameBuilder overwrites the income with what everyone else gets... except they *also* keep track of the original income they had (from the constructor), so they'll go back to 1k-based income as soon as they use an ability.
I suppose we could move to having an `init` function to call after the GameBuilder sets up the variables, but that's a code smell if the peeps on StackOverflow and such are to be believed.

> Frankly, I'd be more inclined to make `incomePerCity` an attribute of `GameInstance` instead (or of a `Scenario` class that I think we will want eventually), since income is set per-game (not per-CO) anyway.

And then COs could freely play with an `incomeAdjustment` property? That should work just fine... with the caveat that now the COs need access to that object at construction (Bear&Bull), turn init (everyone, unless we move the income logic), and events (Patch).